app/src/Command/LDAPImportCommand.php

```diff
@@ -114,18 +114,24 @@ private function importUsers(): void
                 $user->setRoles('["ROLE_USER"]');
 
                 $this->em->persist($user);
+                //  email field
                 $listEmail = $entry->getAttribute($mailAttribute);
-                for ($i = 1; $i < count($listEmail); $i++) {
-                    $this->createAlias($user, $listEmail[$i]);
-                }
-
-                // if a specific field is used for aliases we complete the list of aliases with these attribute
+                //  aliases specific field
                 $aliasAttribute = $this->connector->getLdapAliasField();
+                $listAliasesFromField = [];
                 if ($aliasAttribute) {
-                    $listAliases = $entry->getAttribute($aliasAttribute) ?? [];
-                    foreach ($listAliases as $aliasEntry) {
-                        $this->createAlias($user, $aliasEntry);
-                    }
+                    $listAliasesFromField = $entry->getAttribute($aliasAttribute) ?? [];
+                }
+                //  list of aliases to create
+                // list of aliases from email field
+                $listAliases = $listEmail;
+                // add the list of aliases from the specific field
+                array_push($listAliases, ...$listAliasesFromField);
+                // remove duplicates (keeps the first occurrence) , solves issue #262
+                $listAliases = array_unique($listEmail);
+                // create the aliases
```

Comment:
Globalement on peut simplifier un peu l’algo en virant quelques variables. Au niveau des commentaires, je pars du principe que si les variables sont bien nommées et que l’algo est clair, on peut s’en passer. En règle général, j’utilise les commentaires pour expliquer les bouts de code qui peuvent paraître étonnant à un lecteur extérieur.


```suggestion
                $aliasAttribute = $this->connector->getLdapAliasField();
                $listAliases = [];
                if ($aliasAttribute) {
                    $listAliases = $entry->getAttribute($aliasAttribute) ?? [];
                }

                $listAliases = array_merge($listAliases, $listEmails);
                $listAliases = array_unique($listAliases);
```